android/java/org/chromium/chrome/browser/crypto_wallet/adapters/WalletCoinAdapter.java

```diff
@@ -285,12 +287,22 @@ public void updateSelectedNetwork(String title, String subTitle) {
     }
 
     private void updateSelectedNetwork(int selectedAccountPosition) {
+        if (selectedAccountPosition == previousSelectedPos) {
+            walletListItemModelList.get(selectedAccountPosition).setIsUserSelected(true);
+            notifyItemChanged(previousSelectedPos);
+            return;
+        }
         walletListItemModelList.get(previousSelectedPos).setIsUserSelected(false);
-        notifyItemChanged(previousSelectedPos);
-
         walletListItemModelList.get(selectedAccountPosition).setIsUserSelected(true);
+
+        if (previousSelectedPos < selectedAccountPosition) {
+            notifyItemRangeChanged(
+                    previousSelectedPos, selectedAccountPosition - previousSelectedPos);
+        } else {
+            notifyItemRangeChanged(
+                    selectedAccountPosition, previousSelectedPos - selectedAccountPosition + 1);
```

Comment:
[Adding the comment here to be marked as resolved]

Unfortunately, this still happens (video at 15 second and at the end), I guess an ideal solution could be to use custom checkbox instead of ImageView. Probably discuss with @srirambv or @jamesmudgett to make sure it's good enough to go or requires more polishing.

https://user-images.githubusercontent.com/11755381/176136085-39d62ab9-2854-4555-a667-a22fe0f8f814.mp4

Reply:
due to account switch transition is a minor p4 issue so would not merge that issue, Will only merge https://github.com/brave/brave-browser/issues/23558 after review.